src/StringManipulate.java

```diff
@@ -0,0 +1,255 @@
+			{
+				start=end;
+				end=loop_var;
+				if(max_end-max_start<=end-start)
+				{
+					max_start=start;
+					max_end=end;
+				}
+			}
+		}
+		if(max_end-max_start==0&&length!=0)
+		{
+			max_end=length;
+		}
+		return subString(max_start,max_end,string);
+	}
+
+	public static String subString(int max_start, int max_end, String string)
+	{
+		/*
+		 * This function returns the substring of a type string for a string.
+		 * The starting and ending point for substring are accepted as input.
+		 * String is build by adding character from start to end location from main string.
+		 * StringBuilder is used for reduction the number of immutable object creation
+		 */
+		StringBuilder sub_string=new StringBuilder();
+		while(max_start<max_end)
+		{
+			sub_string.append(string.charAt(max_start));
+			max_start++;
+		}
+		return sub_string.toString();
+	}
+
+	public static String toggleCase(String string)
+	{
+		/*
+		 * Traverse through the array and pick up characters at the index position.
+		 * Check for the ASCII value of character and append its toggled version to the result.
+		 * 32 is the difference between integral value'A' and 'a', everything else goes unchanged like space symbols etc.
+		 */
+		StringBuilder toggled_string=new StringBuilder();
+		for(int temp=0;temp<getLength(string);temp++)
+		{
+			char ch=string.charAt(temp);
+			if(ch>='A'&&ch<='Z')
+			{
+				toggled_string.append((char)(ch+32));
+			}
+			else if(ch>='a'&&ch<='z')
+			{
+				toggled_string.append((char)(ch-32));
+			}
+			else
+			{
+				toggled_string.append(ch);
+			}
+		}
+		return toggled_string.toString();
+	}
+
+	public static String reverseString(String string)
+	{
+		/*
+		 * It takes string as input and break the characters starting from end of origin string and produce a reverse equivalent of original string
+		 * String builder is used to reduce the creation of mutable objects
+		 */
+		StringBuilder reverse_string=new StringBuilder();
+		for(int temp=getLength(string)-1;temp>=0;temp--)
+		{
+			reverse_string.append(string.charAt(temp));
+		}
+		return reverse_string.toString();
+	}
+
+	public static int compareString(String string_one,String string_two)
+	{
+		/*
+		 * Two strings can be equal only of they have a equal length.
+		 * Return 0 if String1 length is not equal to String2 length.
+		 * Without using Api's string needs to be compared word by word.
+		 * Return 1 if both the string are equals.
+		 */
+		// length is stored as for large string calculating length by custom method can increase time complexity
+		int length = getLength(string_one);
+		if(length==getLength(string_two))
+		{
+			for(int temp=0;temp<length;temp++)
+			{
+				if(string_one.charAt(temp)!=string_two.charAt(temp))
+				{
+					return 0;
+				}
+			}
+		}
+		else
+		{
+			return 0;
+		}
+		/*
+		 * If reached here then both the strings are equals considering case also
+		 */
+		return 1;
+	}
+	public static int getLength(String string)
```

Comment:
Never use keyword as an identifier, even with different case
```suggestion
	public static int getLength(String inputString)
```